### ISSUE TITLE
Deprecate Module for core with 9.3

### ISSUE THREAD

**Issue by timodwhit** [amcgowanca/discoverable_entity_bundle_classes#21]
With the release of 9.3 and this https://www.drupal.org/node/3191609, does it make sense to deprecate this module for core?

**Comment by amcgowanca** [comment]
@timodwhit ,

Yes, potentially - it looks to solve the same issues. I need to spend more time investigating, however.

**Comment by benaboki** [comment]
I inherited a site that uses this module. Do you have any tips that you can share to allow us to safely update the site to the drupal 9.3? We are currently locked at 9.2 because of incompatibility. Thanks for the module!

**Comment by benaboki** [comment]
A specific example of something I need to change is below. I will really appreciate any help to rewrite this for 9.3 

```
use Drupal\discoverable_entity_bundle_classes\Annotation\ContentEntityBundleClass;
use Drupal\discoverable_entity_bundle_classes\ContentEntityBundleInterface;
use Drupal\file\Entity\File;

/**
 *
 * @ContentEntityBundleClass(
 *   label = @Translation("Slideshow"),
 *   entity_type = "node",
 *   bundle = "flexslider_slideshow"
 * );
 */
class FlexsliderSlideshow extends \Drupal\node\Entity\Node implements ContentEntityBundleInterface
{
  public function getSlideImage() : ?File {
    $media = $this->field_slide_image->entity;

```

**Comment by amcgowanca** [comment]
Hey @benaboki , if you want to shoot me a quick email with some additional information, I can see how I can help you out. Email is aaron@digitalconvergence.ca

Thanks!

**Comment by nadsrosun** [comment]
@benaboki Have you been able to overcome the issue?

**Comment by benaboki** [comment]
@nadsrosun  we fixed the issue and managed to upgrade to 9.3.9. I share the steps followed in case someone is stuck, we removed all references to discoverable_entity_bundle_classes:

## Remove discoverable entity bundle classes
- use Drupal\discoverable_entity_bundle_classes\Storage\Node\NodeStorage;
- use Drupal\discoverable_entity_bundle_classes\Storage\Taxonomy\TermStorage;

## Update .module file 
### reference custom bundles
+ use Drupal\custom_module\Entity\Term\Country; 
+ use Drupal\custom_module\Entity\Node\Individual;

## add/change hook function

```
*/**
  * Implements hook_entity_bundle_info_alter().
  */
 function custom_module_entity_bundle_info_alter(&$bundles) {
   // Nodes
   $bundles['node']['individual']['class'] = Individual::class;
   $bundles['node']['notifications']['class'] = Notifications::class;

   // Paragraphs
   $bundles['paragraph']['multilingual_download']['class'] = MultilingualDownload::class;

   // Taxonomy Terms.
   $bundles['taxonomy_term']['countries']['class'] = Country::class; 
 }
```
### replace discoverable entity bundle classes
```
   web/modules/custom/custom_module/src/Entity/Node/Individual.php
   - use Drupal\discoverable_entity_bundle_classes\Annotation\ContentEntityBundleClass;
    - use Drupal\discoverable_entity_bundle_classes\ContentEntityBundleInterface;
    + use Drupal\node\Entity\Node;
    + use Drupal\taxonomy\Entity\Term;
    + use Drupal\Core\Annotation\Translation;
    + use Drupal\ipbes_logic\Entity\Node\IndividualInterface;
    use Drupal\user\Entity\User;
    use Drupal\ipbes_logic\Entity\Term\Country;
```    
    
```
    - class Individual extends \Drupal\node\Entity\Node implements IndividualInterface, ContentEntityBundleInterface {
    + class Individual extends Node implements IndividualInterface {
```

   web/modules/custom/custom_module/src/Entity/Node/IndividualInterface.php
```
   + use Drupal\node\NodeInterface;
   - interface IndividualInterface
   + interface IndividualInterface extends NodeInterface
```
    
    Once we got things working with the new set up we removed the module and upgraded the site.

**Comment by timodwhit** [comment]
Nice write up [benaboki](https://github.com/benaboki). 

I'm going to comment with some formatting that might make it easier to follow: 

If you are using DEBC, here is a review for migrating:

### Update .module
Original: 
```php
/**
 * Implements hook_entity_type_alter().
 */
function mymodule_entity_type_alter(&$entity_types) {
  /** @var \Drupal\Core\Entity\EntityTypeInterface[] $entity_types */
  if (isset($entity_types['node'])) {
    $entity_types['node']->setStorageClass('\Drupal\discoverable_entity_bundle_classes\Storage\Node\NodeStorage');
  }
  if (isset($entity_types['taxonomy_term'])) {
    $entity_types['taxonomy_term']->setStorageClass('\Drupal\discoverable_entity_bundle_classes\Storage\Taxonomy\TermStorage');
  }
}
```

to 

```php
use Drupal\mymodule\Entity\BasicPage;

function mymodule_entity_bundle_info_alter(array &$bundles): void {
  if (isset($bundles['node']['page'])) {
    $bundles['node']['page']['class'] = BasicPage::class;
  }
}
```

### In the Class: 
- Remove the annotation: Ex: 
  ```php 
  /**
   * @ContentEntityBundleClass(
   *   label = @Translation("Page"),
   *   entity_type = "node",
   *   bundle = "page"
   * )
   */ );
   ```
- Remove the  `ContentEntityBundleInterface` from the implements
- Remove the associated `use` statement
